### DISCUSSION TITLE
Expanding variable into classname?

### DISCUSSION THREAD

**Discussion by biodigitalfish** [tailwindlabs/tailwindcss#7133] [Help] [answered]
I'm fairly fresh to web dev. Looking for some help / guidance.
I have a custom animation i'd like to be able to override values for. This animation works fine: 

```js
theme: {
    extend: {
      animation: {
        fadeIn: "fadeIn 300ms ease-in 100ms backwards",
      },
      keyframes: {
        fadeIn: {
          "0%": { opacity: 0 },
          "100%": {
            opacity: 1,
          },
        },
      },
      padding: {
        "1/3": "33.33333%",
        "2/3": "66.66667%",
      },
    },
  },
```

However I'm wanting to be able to override the animation / delay time properties. This is so i can increment the animation time for subsequent elements when mapping through an array.

```js


someList.map((item) => {
  fadeInTime += 20
  const fadeInClass = "animate-[fadeIn_" + fadeInTime + "ms_ease-in_100ms_backwards]"
  console.log("FadeInClass:", fadeInTime)

  return (
    <div key={item} className={`${fadeInClass}`}>
      Element..
    </div>
  )
})

```

console.log output on a couple of iterations to show what the const value is:

```js
FadeInClass: animate-[fadeIn_160ms_ease-in_100ms_backwards] 
FadeInClass: animate-[fadeIn_180ms_ease-in_100ms_backwards] 
FadeInClass: animate-[fadeIn_200ms_ease-in_100ms_backwards]
```

It will work if I manually set the classname e.g. 
`animate-[fadeIn_200ms_ease-in_100ms_backwards]`
But doesn't work when trying to substitute the variable as the classname.
Is this a case of tailwind doesn't support expanding variables into classnames? 

Any ideas how I could achieve what I'm trying to do?

**Comment by RobinMalfait** [comment, marked as answer]
Hey! Thank you for your question!
Much appreciated! 🙏

This kind of dynamic classes is not supported because the classes are formed at **runtime** and Tailwind is a **compiletime** tool. More information on the subject can be found here: 

- https://tailwindcss.com/docs/content-configuration#dynamic-class-names
- https://tailwindcss.com/docs/content-configuration#class-detection-in-depth

**Reply by biodigitalfish** [reply]
Thank you Robin! Makes sense. And thanks for the links :)
